e3sm_to_cmip/util.py

```diff
@@ -52,37 +50,345 @@ def print_message(message, status='error'):
+        for name in files:
+            if name[-3:] != '.nc':
+                continue
+            index = name.find('_')
+            if index != -1 and name[:index] in var_list or 'all' in var_list:
+                filepaths.append(os.path.join(root, name))
 
-    Returns
+    pbar = ProgressBar(maxval=len(filepaths))
+    pbar.start()
+    for idx, filepath in enumerate(filepaths):
+        datafile = cdms2.open(filepath, 'a')
+        try:
+            datafile.e3sm_source_code_doi = str('10.11578/E3SM/dc.20180418.36')
+            datafile.e3sm_paper_reference = str(
+                'https://doi.org/10.1029/2018MS001603')
+            datafile.e3sm_source_code_reference = str(
+                'https://github.com/E3SM-Project/E3SM/releases/tag/v1.0.0')
+            datafile.doe_acknowledgement = str(
+                'This research was supported as part of the Energy Exascale Earth System Model (E3SM) project, funded by the U.S. Department of Energy, Office of Science, Office of Biological and Environmental Research.')
+            datafile.computational_acknowledgement = str(
+                'The data were produced using resources of the National Energy Research Scientific Computing Center, a DOE Office of Science User Facility supported by the Office of Science of the U.S. Department of Energy under Contract No. DE-AC02-05CH11231.')
+            datafile.ncclimo_generation_command = str(
+                """ncclimo --var=${var} -7 --dfl_lvl=1 --no_cll_msr --no_frm_trm --no_stg_grd --yr_srt=1 --yr_end=500 --ypf=500 --map=map_ne30np4_to_cmip6_180x360_aave.20181001.nc """)
+            datafile.ncclimo_version = str('4.7.9')
+            pbar.update(idx)
+        finally:
+            datafile.close()
+    pbar.finish()
+# ------------------------------------------------------------------
+
+
+def find_atm_files(var, path):
+    """
+    Looks in the given path for all files that match that matche VAR_\d{6}_\d{6}.nc
+
+    Params:
+    -------
+        var (str): the name of the variable to look for
+        path (str): the path of the directory to look in
+    Returns:
+    --------
+        files (list(str)): A list of paths to the matching files
+    """
+    contents = os.listdir(path)
+    files = list()
+    pattern = '{}'.format(var) + r'\_\d{6}\_\d{6}.nc'
+    for item in contents:
+        if re.match(pattern=pattern, string=item):
+            files.append(item)
+    return files
+# ------------------------------------------------------------------
+
+
+def find_mpas_files(component, path):
+    """
+    Looks in the path given for MPAS monthly-averaged files
+
+    Params:
+    -------
+        component (str): Either the mpaso or mpassi component name
+        path (str): The path of the directory to search for files in
+    """
+    component = component.lower()
+    contents = os.listdir(path)
+
+    if component in ['mpaso', 'mpassi']:
+
+        pattern = '{}.hist.am.timeSeriesStatsMonthly.'.format(component) + \
```

Comment:
This will work for newer runs but not for pi-control unless you're renaming the files as part of symlinking them.  They are called `mpascice.hist.am...`